fileread_flag.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 Michal Witkowski. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+package flagz
+
+import (
+	"fmt"
+
+	"io/ioutil"
+
+	flag "github.com/spf13/pflag"
+)
+
+// ReadFileFlags parses the flagset to discover all "fileread" flags and evaluates them.
+//
+// By reading and evaluating it means: attempts to read the file and set the value.
+func ReadFileFlags(flagSet *flag.FlagSet) error {
+	var outerErr error
+	flagSet.VisitAll(func(f *flag.Flag) {
+		if frv, ok := f.Value.(*FileReadValue); ok {
+			if err := frv.readFile(); err != nil {
+				outerErr = fmt.Errorf("reading file flag '%v' failed: %v", f.Name, err)
+			}
+		}
+	})
+	return outerErr
+}
+
+// FileReadValue is a flag that wraps another flag and makes it readable from a local file in the filesystem.
+type FileReadValue struct {
+	parentFlagName string
+	filePath       string
+	flagSet        *flag.FlagSet
+}
+
+// FileReadFlag creates a `Flag` that allows you to pass a flag.
+//
+// If defaultFilePath is non empty, the flagz.ReadFileFlags will expect the file to be there.
+func FileReadFlag(flagSet *flag.FlagSet, parentFlagName string, defaultFilePath string) *FileReadValue {
+	dynValue := &FileReadValue{parentFlagName: parentFlagName, filePath: defaultFilePath, flagSet: flagSet}
+	flagSet.VarPF(dynValue,
+		parentFlagName+"_path",
+		"",
+		fmt.Sprintf("Path to read contents to a file to read contents of '%v' from.", parentFlagName))
+	return dynValue
+}
+
+func (f *FileReadValue) String() string {
+	return fmt.Sprintf("fileread_for(%v)", f.parentFlagName)
+}
+
+func (f *FileReadValue) Set(path string) error {
+	f.filePath = path
```

Comment:
How so? AFAICT, flags are parsed in order of appearance on the command line, which seems like the desired semantics.